app/src/main/java/org/oppia/app/player/state/itemviewmodel/NumericInputViewModel.kt

```diff
@@ -1,12 +1,71 @@
 package org.oppia.app.player.state.itemviewmodel
 
+import android.content.Context
+import android.text.Editable
+import android.text.TextWatcher
+import androidx.annotation.StringRes
+import androidx.databinding.Bindable
+import androidx.databinding.Observable
+import androidx.databinding.ObservableField
+import org.oppia.app.R
 import org.oppia.app.model.InteractionObject
 import org.oppia.app.model.UserAnswer
+import org.oppia.app.player.state.answerhandling.AnswerErrorCategory
+import org.oppia.app.player.state.answerhandling.InteractionAnswerErrorReceiver
 import org.oppia.app.player.state.answerhandling.InteractionAnswerHandler
+import org.oppia.domain.util.normalizeWhitespace
 
 /** [ViewModel] for the numeric input interaction. */
-class NumericInputViewModel : StateItemViewModel(ViewType.NUMERIC_INPUT_INTERACTION), InteractionAnswerHandler {
+class NumericInputViewModel(
+  private val context: Context,
+  private val interactionAnswerErrorReceiver: InteractionAnswerErrorReceiver
+) : StateItemViewModel(ViewType.NUMERIC_INPUT_INTERACTION), InteractionAnswerHandler {
   var answerText: CharSequence = ""
+  private var pendingAnswerError: String? = null
+  var errorMessage = ObservableField<String>("")
+  private val invalidCharsRegex = """^[\d\s.-]+$""".toRegex()
```

Comment:
Also to confirm: we do not want to allow scientific notation, correct?

Reply:
Yes, we do not want to allow scientific notation.

Reply:
Is this also true on Oppia web?

Reply:
I was not able to  find any parser for scientific notation in numeric input of oppia web code base.@Sean could you please let me know if iam wrong about the same or if we need to allow scientific notation in android?.

Reply:
Please use @seanlip, not @sean. The latter does not reach me.

We do not have a scientific notation parser in the web codebase and we do not plan to support it.

Reply:
Okay thank you.